pkg/endpoint/regenerationcontext.go

```diff
@@ -53,12 +53,22 @@ func (r DatapathRegenerationLevel) String() string {
 }
 
 func (e *ExternalRegenerationMetadata) toRegenerationContext() *regenerationContext {
+
+	pCtx := e.ParentContext
+	cFunc := e.ParentCancel
+
+	if pCtx == nil {
+		pCtx, cFunc = context.WithCancel(context.Background())
```

Comment:
Good find, I never knew this. IMO this should be handled in `endpoint.Regenerate` since the endpoint's EventQueue can be drained, and the actual event handling logic may never occur for an endpoint regeneration that was queued up. It thus has to be the queuer's responsibility.

Reply:
Should be fixed now :)